spyql/processor.py

```diff
@@ -409,18 +468,25 @@ def get_input_iterator(self):
 
 
 class TextProcessor(Processor):
-    def __init__(self, prs, strings):
-        super().__init__(prs, strings)
+    def __init__(self, prs, strings, interactive, filepath = None):
+        super().__init__(prs, strings, interactive)
+        self.filepath = filepath
 
     # reads a text row as a row with 1 column
     def get_input_iterator(self):
         # to do: suport files
-        return ([line.rstrip("\n\r")] for line in sys.stdin)
+        if self.interactive:
+            with open(self.filepath, "r") as f:
+                for line in f:
+                    # note yeilding to avoid reading in one go
+                    yield [line.rstrip("\n\r")]
+        else:
+            return ([line.rstrip("\n\r")] for line in sys.stdin)
```

Comment:
I would suggest Processors to receive a `input_file` rather than a path.  This file would default to `sys.stdin`. 
This way we would move the boilerplate of file opening to a single place.
Then, reading would simply be:
`return ([line.rstrip("\n\r")] for line in input_file)`

This applies to all processors.